### PR TITLE
refactored the code for cluster handler

### PR DESCRIPTION
**Pull request by vklohiya**
**Description**:  _Add Issue/Feature description_

**Changes Proposed in PR**:

**Fixes**: resolves #_Github issue id_

## General Checklist
- [x] Updated Added functionality/ bug fix in release notes
- [x] Added examples for new feature
- [x] Updated the documentation
- [x] Smoke testing completed

## CRD Checklist
- [x] Updated required CR schema

### CODE REVIEW

**Review by charanm08 — APPROVED** [review]
LGTM

**Review by lavanya-f5 — APPROVED** [review]
LGTM

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -189,27 +188,30 @@ func NewController(params Params, startController bool) *Controller {
...
+				clusterConfig.namespaces[ns] = true
 			}
 		}
 	} else {
-		err2 := ctlr.createNamespaceLabeledInformerForCluster(ctlr.multiClusterConfigs.ClusterConfigs[""].namespaceLabel, "")
+		err2 := ctlr.createNamespaceLabeledInformerForCluster(params.NamespaceLabel, "")
 		if err2 != nil {
 			log.Errorf("%v", err2)
-			for _, nsInf := range ctlr.multiClusterConfigs.ClusterInformers[""].nsInformers {
+			informerStore := ctlr.multiClusterConfigs.getInformerStore("")
```
Comment (pmahdev) [inline pkg/controller/controller.go:210]: What if there is no informer for that cluster, can we add a safety check 

`informerStore`

File: pkg/controller/clusterHandler.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+// enqueueEvent adds an event to the eventQueue after checking for uniqueness.
+func (ch *ClusterHandler) enqueueEvent(clusterName string, obj interface{}) {
+	key := fmt.Sprintf("%s/%s", clusterName, "")
+	if _, exists := ch.uniqueAppIdentifier[key]; exists {
+		fmt.Printf("Duplicate event discarded: %s\n", key)
+		return
+	}
+	ch.uniqueAppIdentifier[key] = struct{}{}
+	ch.eventQueue.Add(obj)
+	fmt.Printf("Event queued: %s\n", key)
```
Comment (pmahdev) [inline pkg/controller/clusterHandler.go:91]: Is it good to add the logger functions instead of `fmt` ?

File: pkg/controller/controller.go (hunk 1 above)
Comment (vklohiya) [reply pkg/controller/controller.go:210]: Yeah, Code will be further refactored in upcoming PRs.

File: pkg/controller/clusterHandler.go (hunk 2 above)
Comment (vklohiya) [reply pkg/controller/clusterHandler.go:91]: This function is a placeholder for now.

File: pkg/controller/controller.go (hunk 1 above)
Comment (pmahdev) [reply pkg/controller/controller.go:210]: Ok, Got it~

File: pkg/controller/clusterHandler.go (hunk 2 above)
Comment (pmahdev) [reply pkg/controller/clusterHandler.go:91]: OK

**Comment by pmahdev** [comment]
LGTM

**Review by arzzon — APPROVED** [review]
LGTM

**Pull request merged**